state/azure/cosmosdb/cosmosdb.go

```diff
@@ -335,23 +359,46 @@ func (c *StateStore) Multi(request *state.TransactionalStateRequest) error {
 	return nil
 }
 
-// This is a helper to return the partition key to use.  If if metadata["partitionkey"] is present,
-// use that, otherwise use what's in "key".
-func populatePartitionMetadata(key string, requestMetadata map[string]string) string {
-	if val, found := requestMetadata[metadataPartitionKey]; found {
-		return val
+func createUpsertItem(contentType string, req state.SetRequest, partitionKey string) CosmosItem {
+	byteArray, isBinary := req.Value.([]uint8)
+	if isBinary {
+		if contenttype.IsJSONContentType(contentType) {
+			var value map[string]interface{}
+			err := json.Unmarshal(byteArray, &value)
+			// if byte array is not a valid JSON, so keep it as-is to be Base64 encoded in CosmosDB.
+			// otherwise, we save it as JSON
+			if err == nil {
+				return CosmosItem{
+					ID:           req.Key,
+					Value:        value,
+					PartitionKey: partitionKey,
+					IsBinary:     false,
+				}
+			}
```

Comment:
i think err is being swallowed here. If it is not nil(improper json), it should be returned to the caller?

Reply:
This is by design. In case the data is not valid JSON, we write it as binary.